x/ibc/testing/chain.go

```diff
@@ -0,0 +1,373 @@
+// and the height at which the query was performed.
+func (chain *TestChain) QueryProof(key []byte) (commitmenttypes.MerkleProof, uint64) {
+	res := chain.App.Query(abci.RequestQuery{
+		Path:   fmt.Sprintf("store/%s/key", host.StoreKey),
+		Height: chain.App.LastBlockHeight(),
+		Data:   key,
+		Prove:  true,
+	})
+
+	proof := commitmenttypes.MerkleProof{
+		Proof: res.Proof,
+	}
+
+	return proof, uint64(res.Height)
+}
+
+// nextBlock sets the last header to the current header and increments the current header to be
+// at the next block height. It does not update the time as that is handled by the IBCTestSuite.
+//
+// CONTRACT: this function must only be called after app.Commit() occurs
+func nextBlock(chain *TestChain) {
+	// set the last header to the current header
+	chain.LastHeader = ibctmtypes.CreateTestHeader(
+		chain.CurrentHeader.ChainID,
+		chain.CurrentHeader.Height,
+		chain.CurrentHeader.Time,
+		chain.Vals, chain.Signers,
+	)
+
+	// increment the current header
+	chain.CurrentHeader = abci.Header{
+		Height: chain.CurrentHeader.Height + 1,
+		Time:   chain.CurrentHeader.Time,
+	}
+}
+
+// SendMsg delivers a transaction through the application. It updates the senders sequence
+// number and updates the TestChain's headers.
+func (chain *TestChain) SendMsg(msg sdk.Msg) {
+	_, _, err := simapp.SignCheckDeliver(
+		chain.t,
+		chain.App.Codec(),
+		chain.App.BaseApp,
+		chain.GetContext().BlockHeader(),
+		[]sdk.Msg{msg},
+		[]uint64{chain.SenderAccount.GetAccountNumber()},
+		[]uint64{chain.SenderAccount.GetSequence()},
+		true, true, chain.senderPrivKey,
+	)
+	require.NoError(chain.t, err)
+
+	// SignCheckDeliver calls app.Commit()
+	nextBlock(chain)
+
+	// increment sequence for successful transaction execution
+	chain.SenderAccount.SetSequence(chain.SenderAccount.GetSequence() + 1)
+}
+
+// NewConnection appends a new connectionID string in the format:
+// connectionID<index>
+func (chain *TestChain) NewConnection() string {
+	conn := ConnectionIDPrefix + string(len(chain.Connections))
+
+	chain.Connections = append(chain.Connections, conn)
+	return conn
+}
+
+// NewChannel appends a new testing channel which contains references to the port and channel ID
+// used for channel creation and interaction.
+func (chain *TestChain) NewChannel() Channel {
+	portID := PortIDPrefix + string(len(chain.Channels))
+	channelID := ChannelIDPrefix + string(len(chain.Channels))
+	channel := NewChannel(portID, channelID)
+
+	chain.Channels = append(chain.Channels, channel)
+	return channel
+}
+
+// CreateClient will construct and execute a 07-tendermint MsgCreateClient. A counterparty
+// client will be created on the (target) chain.
+func (chain *TestChain) CreateClient(counterparty *TestChain) {
+	// construct MsgCreateClient using counterparty
+	msg := ibctmtypes.NewMsgCreateClient(
+		counterparty.ClientID, counterparty.LastHeader,
+		lite.DefaultTrustLevel, TrustingPeriod, UnbondingPeriod, MaxClockDrift,
```

Comment:
`lite.DefaultTrustLevel` should also be a variable defined at the top level of the file so that it can be overwritten in a case it's necessary